IRuettae/Core/IRuettae.Core.LocalSolver/Solver.cs

```diff
@@ -0,0 +1,418 @@
+                {
+                    visitSequences[k] = model.List(numberOfVisits);
+                    santaOvertime[k] = model.Int(0, int.MaxValue);
+                    santaWaitBeforeStart[k] = model.Int(0, int.MaxValue);
+                    if (UseWaitBetweenVisits)
+                    {
+                        santaWaitBetweenVisit[k] = new LSExpression[visits.Count];
+                        for (int i = 0; i < santaWaitBetweenVisit[k].Length; i++)
+                        {
+                            santaWaitBetweenVisit[k][i] = model.Int(0, int.MaxValue);
+                        }
+                    }
+                }
+
+                // overflow for unused santa breaks
+                visitSequences[numberOfSantas] = model.List(numberOfVisits);
+
+
+                model.Constraint(model.Partition(visitSequences));
+
+                var distanceArray = model.Array(routeCostJagged);
+                var distanceFromHomeArray = model.Array(visits.Select(v => v.WayCostFromHome).ToArray());
+                var distanceToHomeArray = model.Array(visits.Select(v => v.WayCostToHome).ToArray());
+                var visitDurationArray = model.Array(visits.Select(v => v.Duration).ToArray());
+
+                // desired
+                var visitsOnlyDesired = visits
+                    .Select(v =>
+                        // fake arr
+                        v.Desired.Length == 0 ? new[] { new[] { -1, -1 } } :
+                        v.Desired.Select(d => new[] { d.from, d.to }).ToArray()
+                        )
+                    .ToArray();
+                var visitDesiredArray = model.Array(visitsOnlyDesired);
+                var visitDesiredCountArray = model.Array(visits.Select(v => v.Desired.Length).ToArray());
+
+                // unavailable
+                var visitsOnlyUnavailable = visits
+                    .Select(v =>
+                        // fake arr
+                        v.Unavailable.Length == 0 ? new[] { new[] { -1, -1 } } :
+                        v.Unavailable.Select(d => new[] { d.from, d.to }).ToArray()
+                    )
+                    .ToArray();
+                var visitUnavailableArray = model.Array(visitsOnlyUnavailable);
+                var visitUnavailableCountArray = model.Array(visits.Select(v => v.Unavailable.Length).ToArray());
+
+                // ********************
+                // solving
+                // ********************
+                consoleProgress?.Invoke(this, "Begin solving");
+
+                for (int i = 0; i < visits.Count; i++)
+                {
+                    if (!visits[i].IsBreak)
+                    {
+                        model.AddConstraint(!model.Contains(visitSequences[numberOfSantas], i));
+                    }
+                }
+
+                for (int day = 0; day < numberOfDays; day++)
+                {
+                    for (int santa = 0; santa < input.Santas.Length + numberOfFakeSantas; santa++)
```

Comment:
single source of truth: Konstante (numberOfSantas) verwenden.
Weiter unten dasselbe.

Reply:
ist nicht dasselbe, numberOfSantas ist eig. number of routes (santa + fakesanta)*numberOfDays

Reply:
hab numberOfSantas auf numberOfRoutes umbenannt um verwirrung zu vermeiden

Reply:
passt.